galaxy/main/signals/handlers.py

```diff
@@ -41,15 +41,17 @@ def user_logged_in_handler(request, user, **kwargs):
     user.avatar_url = social.extra_data.get('avatar_url')
     user.save()
 
-    if user.namespaces.count():
-        # User is associated with one or more namespaces.
-        # Even though the Namespaces may be inactive, we'll assume
-        # the user has been here before, and knows how to manage Namespaces.
+    username = social.extra_data.get('login')
+    sanitized_username = username.lower().replace('-', '_')
+
+    try:
+        namespace = models.ProviderNamespace.objects.get(name=username)
```

Comment:
Think the filter should be name__iexact=username, to account for existing namespaces that may have contain uppercase letters.